### DISCUSSION TITLE
Why am I required to return a field in this resolver?

### DISCUSSION THREAD

**Discussion by vedantroy** [dotansimha/graphql-code-generator#6232] [Q&A]
I have the following snippet in my graphql schema:

```graphql
type Cost {
    id: ID!
    multiplier: Float!
}

type Fruit {
    id: ID!
    cost: Cost
}
```

In my resolvers object, I have the following:
```
Fruit: {
	cost: async (parent, args, context, info) => {
		return { id: parent.id }
	},
}
```

However, Typescript complains and says I also need to include the `multiplier` field in the returned object. On one hand this makes b/c `multiplier` is required. But I'm also confused, shouldn't graphql go look for a `Cost` object and then use a resolver there instead?

**Comment by dotansimha** [comment]
By default, codegen generates a signature that requires you to return what the schema exposes. So if the field is non-nullable, you'll have to return it. 
If you want to customize that, you can use `mappers`, take a look at this: https://the-guild.dev/blog/better-type-safety-for-resolvers-with-graphql-codegen
